scripts/spike.js

```diff
@@ -0,0 +1,228 @@
+        domInteractive: 540,
+        domComplete: 1350
+      }
+    },
+    span_count: {
+      started: 82
+    },
+    sampled: true,
+    experience: {
+      tbt: 100,
+      cls: 0.5,
+      fid: 50
+    }
+  }
+}
+
+const defaultSpan = {
+  span: {
+    id: '8cabde4612857a8f',
+    transaction_id: '93a0bcc9a6a4a646',
+    parent_id: '93a0bcc9a6a4a646',
+    trace_id: '1acfdeb11a825ddecedf54dcce00994c',
+    name: 'http://testing.com',
+    type: 'resource',
+    subtype: 'script',
+    start: 25,
+    duration: 143,
+    context: {
+      http: {
+        url: 'http://testing.com',
+        response: {}
+      },
+      destination: {
+        service: {
+          name: 'http://testing.com',
+          resource: 'testing.com:80',
+          type: 'resource'
+        },
+        address: 'testing.com',
+        port: 80
+      }
+    }
+  }
+}
+
+function getRandomNumber(min, max) {
+  return Math.random() * (max - min) + min
+}
+
+function getRandomInt(min, max) {
+  min = Math.ceil(min)
+  max = Math.floor(max)
+  return Math.floor(Math.random() * (max - min + 1)) + min
+}
+
+function generateTransaction(spanCount) {
+  let tr = merge({}, defaultTransaction, {
+    transaction: {
+      id: generateRandomId(16),
+      trace_id: generateRandomId(),
+      duration: Math.floor(getRandomNumber(0, 5000)),
+      name: faker.random.words(3),
+      span_count: {
+        started: spanCount
+      },
+      sampled: spanCount != 0,
+      context: {
+        page: {
+          referer: faker.internet.url(),
+          url: faker.internet.url()
+        }
+      }
+    }
+  })
+  let payload = [tr]
+  for (let i = 0; i < spanCount; i++) {
+    const span = merge({}, defaultSpan, {
+      span: {
+        id: generateRandomId(16),
+        transaction_id: tr.transaction.id,
+        parent_id: tr.transaction.id,
+        trace_id: tr.transaction.trace_id,
+        name: [
+          'http://testing.com',
+          generateRandomId(),
+          generateRandomId() + '.js'
+        ].join('/'),
+        start: getRandomInt(0, tr.transaction.duration),
+        duration: getRandomInt(0, tr.transaction.duration)
+      }
+    })
+    payload.push(span)
+  }
+
+  return payload
+}
+
+function ndJsonStringify(object) {
+  return JSON.stringify(object) + '\n'
+}
+
+async function postPayload(url, payload) {
+  let data = payload.map(p => {
+    return ndJsonStringify(p)
+  })
+  let response = await fetch(url, {
+    method: 'POST',
+    cache: 'no-cache',
+    headers: {
+      'Content-Type': 'application/x-ndjson',
+      'X-Forwarded-For': faker.internet.ip(),
+      'User-Agent': faker.internet.userAgent()
+    },
+    body: data.join('')
+  })
+  return response.text()
+}
+
+async function generatePayloads(trCount) {
+  let promises = []
+  for (let i = 0; i < trCount; i++) {
+    let payload = generateTransaction(50)
+    payload.unshift(merge({}, defaultMeta))
+    let p = postPayload('http://localhost:8200/intake/v2/rum/events', payload)
```

Comment:
we can use our test config setup to get the server url https://github.com/elastic/apm-agent-rum-js/blob/e4e10d5b7c451a950808d605a2db99cd275d20eb/dev-utils/test-config.js#L45